services/system/StagedTx/service/src/lib.rs

```diff
@@ -0,0 +1,440 @@
+    /// * `txid`: The unique txid of the staged transaction
+    #[action]
+    fn accept(id: u32, txid: [u8; 32]) {
+        let staged_tx = StagedTx::get(id, txid);
+
+        Response::upsert(id, true);
+
+        staged_tx
+            .staged_tx_policy()
+            .accept(staged_tx.id, get_sender());
+
+        staged_tx.emit(StagedTxEvent::ACCEPTED);
+    }
+
+    /// Indicates that the caller rejects the staged transaction
+    ///
+    /// * `id`: The ID of the database record containing the staged transaction
+    /// * `txid`: The unique txid of the staged transaction
+    #[action]
+    fn reject(id: u32, txid: [u8; 32]) {
+        let staged_tx = StagedTx::get(id, txid);
+
+        Response::upsert(id, false);
+
+        staged_tx
+            .staged_tx_policy()
+            .reject(staged_tx.id, get_sender());
+
+        staged_tx.emit(StagedTxEvent::REJECTED);
+    }
+
+    /// Removes (deletes) a staged transaction
+    ///
+    /// A staged transaction can only be removed by the proposer or the auth service of
+    /// the staged tx sender.
+    ///
+    /// * `id`: The ID of the database record containing the staged transaction
+    /// * `txid`: The unique txid of the staged transaction
+    #[action]
+    fn remove(id: u32, txid: [u8; 32]) {
+        let staged_tx = StagedTx::get(id, txid);
+
+        check(
+            get_sender() == staged_tx.proposer || get_sender() == staged_tx.first_sender(),
+            "Only the proposer or the staged tx first sender can remove the staged tx",
+        );
+
+        // Delete all responses for this staged tx
+        let responses = ResponseTable::new();
+        let deletable = responses
+            .get_index_pk()
+            .range((id, AccountNumber::new(0))..=(id, AccountNumber::new(u64::MAX)))
+            .collect::<Vec<_>>();
+        for d in deletable {
+            responses.erase(&(d.id, d.account));
+        }
+
+        // Delete the staged tx itself
+        StagedTxTable::new().erase(&id);
+
+        staged_tx.emit(StagedTxEvent::DELETED);
+    }
+
+    /// Notifies the staged-tx service that a staged transaction should be executed.
+    /// Typically this call is facilitated by the staged transaction's first sender's auth
+    /// service on behalf of the staged transaction's first sender.
+    ///
+    /// * `id`: The ID of the database record containing the staged transaction
+    /// * `txid`: The unique txid of the staged transaction
+    #[action]
+    fn execute(id: u32, txid: [u8; 32]) {
+        let staged_tx = StagedTx::get(id, txid);
+
+        check(
+            staged_tx.first_sender() == get_sender(),
+            "Only the first sender of the staged tx can execute it",
+        );
+
+        staged_tx.action_list.actions.iter().for_each(|action| {
+            let _ = Transact::call().runAs(action.clone(), Vec::new());
+        });
+
+        staged_tx.emit(StagedTxEvent::EXECUTED);
+
+        remove(id, txid);
```

Comment:
`remove` should happen before running the actions to prevent multiple execution.

Reply:
done